lib/browser/api/menu.js

```diff
@@ -43,63 +43,50 @@ Menu.prototype._init = function () {
   }
 }
 
-Menu.prototype.popup = function (window, x, y, positioningItem) {
-  let [newX, newY, newPosition, newWindow] = [x, y, positioningItem, window]
-  let opts
-  let callback
-
-  // menu.popup(x, y, positioningItem)
-  if (window != null && !(window instanceof BrowserWindow)) {
-    [newPosition, newY, newX, newWindow] = [y, x, window, null]
+Menu.prototype.popup = function (window, options, callback) {
+  let x, y, pos
+  let [win, opts, cb] = [window, options, callback]
+
+  if (!opts && !callback) {
+    // win.popup({opts})
+    if (typeof window !== 'function') {
+      opts = window
+    // win.popup(callback)
+    } else {
+      callback = window
+    }
```

Comment:
Okay, at a very shallow level I'd recommend inverting this logic s.t. it reads

```js
if (typeof window === 'function') {
  callback = window
} else {
  opts = window
}
```

But my bigger question is why the code needs to inspect the type of these arguments in the first place. Are there multiple use cases where you can just call `popup(window, {foo: bar}, callback)` and just `popup(callback)`? That latter signature isn't listed in the documentation and that kind of bouncing back and forth makes this code confusing, e.g. this function allows almost any combination of window, options, and callback.

If there is not a use case for all these combinations, this code could be simplified by testing all these combinations.

If there is a use case, maybe they should be documented?